java/src/main/java/com/powsybl/python/network/NetworkModificationCFunctions.java

```diff
@@ -0,0 +1,89 @@
+import com.powsybl.python.commons.CTypeUtil;
+import com.powsybl.python.commons.Directives;
+import com.powsybl.python.commons.PyPowsyblApiHeader;
+import org.apache.commons.lang3.Range;
+import org.graalvm.nativeimage.IsolateThread;
+import org.graalvm.nativeimage.ObjectHandle;
+import org.graalvm.nativeimage.ObjectHandles;
+import org.graalvm.nativeimage.c.CContext;
+import org.graalvm.nativeimage.c.function.CEntryPoint;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.CIntPointer;
+
+import java.util.*;
+
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.*;
+import static com.powsybl.python.commons.Util.createIntegerArray;
+import static com.powsybl.python.commons.Util.doCatch;
+
+/**
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+@CContext(Directives.class)
+public final class NetworkModificationCFunctions {
+
+    private NetworkModificationCFunctions() {
+    }
+
+    @CEntryPoint(name = "getConnectablesOrderPositions")
+    public static PyPowsyblApiHeader.ArrayPointer<PyPowsyblApiHeader.SeriesPointer> getConnectablesOrderPositions(IsolateThread thread, ObjectHandle networkHandle,
+                                                     CCharPointer voltageLevelId, PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            String voltageLevelIdStr = CTypeUtil.toString(voltageLevelId);
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+
+            VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelIdStr);
+            Map<String, Integer> feederPositionsOrders = getFeederPositionsByConnectable(voltageLevel);
+            return Dataframes.createCDataframe(Dataframes.mapMapper(), feederPositionsOrders);
+        });
+
+    }
+
+    @CEntryPoint(name = "getUnusedConnectableOrderPositionsBefore")
+    public static PyPowsyblApiHeader.ArrayPointer<CIntPointer> getUnusedConnectableOrderPositionsBefore(IsolateThread thread, ObjectHandle networkHandle,
+                                                                                             CCharPointer busbarSectionId,
+                                                                                             PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        Network network = ObjectHandles.getGlobal().get(networkHandle);
+        String busbarSectionIdStr = CTypeUtil.toString(busbarSectionId);
+        BusbarSection busbarSection = network.getBusbarSection(busbarSectionIdStr);
+        Optional<Range<Integer>> positionsOrders = getUnusedOrderPositionsBefore(busbarSection);
+        if (positionsOrders.isPresent()) {
+            int max = positionsOrders.get().getMaximum();
+            int min = positionsOrders.get().getMinimum();
+            return createIntegerArray(Arrays.asList(min, max));
+        } else {
+            //TODO: It actually doesn't work
```

Comment:
In this function and the one below, I am not sure what should be returned in the case where there are no extensions. The empty list doesn't work with the C++/python. Any idea?